rero_ils/modules/users/views.py

```diff
@@ -42,6 +44,24 @@ def is_logged_librarian(*args, **kwargs):
     return is_logged_librarian
 
 
+def check_user_permission(fn):
+    """Decorate to check permission access.
+
+    The access is allow when the connected user is a librarian or
+    the user id is the same of the id argument.
+    """
+    @wraps(fn)
+    def is_logged(*args, **kwargs):
+        """Decorated view."""
+        if not current_user.is_authenticated:
+            abort(401)
+        if not current_librarian and 'id' in kwargs \
+                and current_user.id != int(kwargs.get('id')):
+            abort(401)
+        return fn(*args, **kwargs)
+    return is_logged
+
+
```

Comment:
NOTE: It should be possible to work with Invenio `Permission`. Maybe to implement after "permission spike US"